addons/udes_stock/models/models.py

```diff
@@ -45,3 +48,48 @@ def base_model_detailed_view(self, model, form_view):
             "res_id": self.id,
             "context": {"view_all_fields": True},
         }
+
+    @api.model
+    def _create(self, vals):
+        """Inheriting low level _create method to check if there is any date or datetime field less
+        than a static year 1000. In that case raise an error to show the users that the date entered
+         is not correct"""
+        date_fields = self.get_date_or_datetime_fields()
+        if isinstance(vals, dict):
+            self.validate_values_of_date_fields(vals, date_fields)
+        res = super()._create(vals)
+        return res
+
+    @api.multi
+    def _write(self, vals):
+        """Inheriting low level _write method to check if there is any date or datetime field less
+        than a static year 1000. In that case raise an error to show the users that the date entered
+         is not correct"""
+
+        date_fields = self.get_date_or_datetime_fields()
+        if isinstance(vals, dict):
+            self.validate_values_of_date_fields(vals, date_fields)
+        res = super()._write(vals)
+        return res
+
+    @api.model
+    def validate_values_of_date_fields(self, values, date_fields):
+        """
+        Checking if date or datetime fields that are formatted as string
+        can be converted to a date field and the year is greater than 1000.
+        Raising error in case year is less than 1000
+        """
+        for key, value in values.items():
+            if key in date_fields:
+                if isinstance(value, str) and len(value) >= 10:
+                    date_field = datetime.strptime(value[:10], DATE_FORMAT)
+                    if date_field.year < 1000:
+                        raise UserError(
+                            _("Year of %s is less than 1000") % value
+                        )
+
+    def get_date_or_datetime_fields(self):
+        """Finding date or datetime fields of current model"""
+        return {
```

Comment:
Suggestion: Don't include `create_date` and `write_date` as these will be set by the system, and we can minimise any potential performance hits.